sophia-vote-contract.md

```diff
@@ -100,37 +106,37 @@ contract Vote =
    record state = {
       votes: map(address, candidates) }
 
-   public stateful function init() : state = {
+   stateful entrypoint init() : state = {
       votes = { } }
 
-   public stateful function vote(candidate: address) : bool =
+   stateful entrypoint vote(candidate: address) : bool =
       is_candidate(candidate)
```

Comment:
This is convoluted, you better use a default value when looking up the candidate and do away with the `is_candidate` function. I.e. something like this:
```
stateful vote(candidate: address) : unit = 
  let current_votes = state.votes[candidate = {voters = []}].voters
  put(state{ votes[candidate] = {voters = Call.caller :: current_votes }})
```

Reply:
I have a dilemma here. With this change, there will be no need for `add_candidate` function. Can you give me feedback on the new full contract? I will change the wordings in the tutorial when the contract is satisfactory.

cc: @thepiwo ,